### PR TITLE
midproject

### PR DESCRIPTION
**Pull request by zhiyangcheniris**
Hello @latterfate!

I have made the following changes to this repo:

- Updated `README.md` with instructions on managing the database (Docker Image, S3 & RDS).
- Added a Dockerfile for building image.
- Updated `.gitignore` file so that local configs and some other testing files are all ignored.
- Updated `requirement.txt` to resolve conflict in packages that existed previously.
- Added module (`src/s3.py`) for uploading to and downloading from my S3 bucket.
- Added module (`src/rds.py`) for connecting to the RDS MySQL database as well as for creating SQLite DB locally, with the structure of recipe table.
- Updated `run.py` so that it can now call both s3 and rds operations. Also, the imports are reformatted so that the imports were at the top before defining the logger (see point below).
- Updated the local logger configuration file (`config/logging/local.conf`) so that the configurations would apply to logs within modules in src/ while still being able to place the imports at the top of `run.py` before defining the logger. Also, configurations for python packages with excessive logs are set to WARNING level. The level of the rest logging is all set to INFO for now.
- Added a config for production  (`config/logging/prod.conf`). It is not used for now, but will be needed in the future.
- Added sample data for the raw dataset for testing purpose (`data/sample/*`).

Please review the files mentioned above and build a Docker image to run the script, following my README.md instructions. 

Thank you!
Iris

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,125 +99,220 @@
...
 
-You can also define the absolute path with four `////`, for example:
+The default `local_path` are <br>
+`data/sample/sample_RAW_recipes.csv` and <br>
+`data/sample/sample_RAW_interactions.csv`.
 
-```python
-engine_string = 'sqlite://///Users/cmawer/Repos/2020-MSIA423-template-repository/data/tracks.db'
+#### 3) Download Raw Datasets from S3 Bucket
+To upload datasets, please run:
```
Comment (latterfate) [inline README.md:151]: typo: to download datasets

File: README.md [hunk 2]
```diff
@@ -63,125 +99,220 @@
...
+
+```bash
+docker run -e AWS_ACCESS_KEY_ID -e AWS_SECRET_ACCESS_KEY recipe run.py s3 --download\
```
Comment (latterfate) [inline README.md:154]: for windows users, add winpty in the front

**Review by latterfate — COMMENTED** [review]
Hi Zhiyang,

Sorry for doing the review in last minute. Overall, you have done a fantastic job and there is not many suggestions that I can give you.

For functionality and readability, I think you have written very clear and efficient code. I followed your instructions at readme and find everything to be self-explanatory. There is one or two typos that I marked out but I am still very impressed by the quality of work overall.

The overall design is very consistent with what we study in the class. I believe the code is very extensible that any people can find it easy enough to contribute more content based on the current structure you set up. 

In terms of documentations, you have made a lot of explanations inside every step and I am just impressed by the time and effort you have spent to achieve this.

File: README.md (hunk 1 above)
Comment (zhiyangcheniris) [reply README.md]: Thanks for catching that! I have fixed it!

File: README.md (hunk 2 above)
Comment (zhiyangcheniris) [reply README.md:154]: I'm not familiar with Windows, and thus I wonder is it the case that they need to add `winpty` at the front for all `docker run` command? If so, I can for sure add a notice at the very beginning!

**Pull request merged**